### PR TITLE
refactor: Added normalize_column_name parameter to HANABaseDialect so…

### PR DESCRIPTION
**Pull request by dabla**
Added a parameter to the HANABaseDialect so that we can configure if we want to normalize column names or not.

At our enterprise, HANA seems to be case sensitive and when we dynamically generate SQL statments based on the columns names returned from the dialect, those do not correspond with the case in our database and thus HANA gives an error saying it doesn't know the column because it's in lower case and in in our case it should be in upper case.

This pull request was based on tag v0.5.0 as we are using Airflow which uses an older version of sqlalchemy which only works with sqlalchemy-hana 0.5.0.  So it would be cool if we could get an additional 0.6.0 version.  I'm also willing to merge this is the current branch so that newer versions of sqlalchemy-hana also benefit from this modification.

So I would need a branch based on 0.5.0 to be able to apply this fix, you can do this in git: `git checkout tags/v0.5.0 -b v0.6.0`

At the moment we have to monkey-patch the get_columns method of the HANABaseDialect in python to bypass this behaviou in sqlalchemy-hana to make it work, so it would be cool if this could become a configurable option.  I also added 2 tests which tests the result of the get_columns method with the default behaviour and when normalize_column_name option is disabled.

### CODE REVIEW

**Comment by kasium** [comment]
Hi, thanks a lot for the contribution. Let me follow up on this next week. I would like to check some things beforehand. Have a nice weekend 👍

**Comment by dabla** [comment]
> Hi, thanks a lot for the contribution. Let me follow up on this next week. I would like to check some things beforehand. Have a nice weekend 👍

Hello, thank you for your answer, I applied to same for the main branch in a second pull request.

**Comment by kasium** [comment]
@dabla Before discussing a feature request, can you please give some details about the issue itself. What is the column name in the database (TABLE_COLUMNS system view), what did you configure in SQLite and what is finally rendered by the dialect. A reproducible example would help a lot

**Comment by dabla** [comment]
> @dabla Before discussing a feature request, can you please give some details about the issue itself. What is the column name in the database (TABLE_COLUMNS system view), what did you configure in SQLite and what is finally rendered by the dialect. A reproducible example would help a lot

I've added 2 tests which reproduce and show what the issue is here through a simple example: https://github.com/SAP/sqlalchemy-hana/pull/205/commits/0d1dc20477457f4b90af31a1dc2da5dd72232e94

I don't need sqlite to be able to reproduce it, I just mocked the whole resultset that I normally receive from HANA.

**Comment by kasium** [comment]
Can you please provide the DDL statement to create the EMPLOYEES table?

**Comment by dabla** [comment]
> Can you please provide the DDL statement to create the EMPLOYEES table?

CREATE COLUMN TABLE "HR"."EMPLOYEES"(
     ID BIGINT CS_FIXED GENERATED BY DEFAULT AS IDENTITY NOT NULL,
	 NAME NVARCHAR(255),
	 FIRST_NAME NVARCHAR(255),
	 GENDER NVARCHAR(1),
     BIRTHDATE DATE,
	)
;

**Comment by kasium** [comment]
But this is exactly the behavior which is intended and needed by SQLAlchemy. See [here](https://github.com/SAP/sqlalchemy-hana?tab=readme-ov-file#case-sensitivity). In short, SAP HANA represents case-insensitive names as uppercase, while SQLAlchemy does this as lowercase. Therefore you will get lowercase names when you use getColumns.

I see the issue in you dynamic SQL generation: You wrap the names in quotes which makes the name case sensitive, therefore SAP HANA will only e.g. match "myColumn" and not "MYCOLUMN". However, if you leave the quotes away, HANA will match the column as intended because behind the scenes inside the SQL it's transformed to uppercase. So either you leave the quotes aside or you transform the column name fetched from SQLAlchemy to uppercase beforehand if it only contains lowercase characters. 

TL;DR;  We won't accept your PR because it adds a wrong behavior. The function getColumns and others are meant to be used for SQLAlchemy and don't necessary represent the native database data. If you need this, you can just execute a native SQL or use on of the ways described above. However, if you would use the parameters suggested by our PR and e.g. a SQLAlchemy reflection, it might will run nuts.

**Comment by dabla** [comment]
Alright, thank you for you comment, I didn't know this behaviour as there was told to always put column names between quotes in HANA, so that might explain it, I will try it and let you know.

**Comment by dabla** [comment]
Hello @kasium , you're right and I also got confirmation by other colleagues that indeed I should not use the quotes which solves my issue.  Thanks again!

**Comment by kasium** [comment]
Great👍
